static/js/src/services/code.js

```diff
@@ -1,29 +1,52 @@
 'use strict';
 
-function code(commands, blockly) {
-  return { generate, execute };
+function code(commands, blockly, Boards) {
+  return { generate, execute, setWorkshop };
+  var currentWorkshop = null;
+
+  // TODO - this seems pretty wrong. Let's ask Brian or someone who knows Angular
+  // pretty well to take a look!
+  function setWorkshop(workshop) {
+    currentWorkshop = workshop;
+  }
```

Comment:
That seems like a more sensible approach! I'll get that change made. Thanks for looking over it and making the recommendation!